docs/source/configuration_file.rst

```diff
@@ -302,9 +302,12 @@ UNet3D (Optional)
 Testing parameters
 ------------------
 
--  ``binarize_prediction``: Bool. Binarize output predictions using a
-   threshold of 0.5. If ``false``, output predictions are float between
-   0 and 1.
+- ``binarize_predictions``: Float. Threshold (between 0 and 1) used to binarize
+    the predictions before computing the validation metrics. To use soft predictions
+    (i.e. no binarisation, float between 0 and 1) for metric computation, indicate -1.
+-  ``binarize_niftis``: Bool. Binarize output predictions using ``binarize_predictions``
```

Comment:
Why do we need this new parameter `binarize_niftis`? If binarize_predicitons is set to `-1` the soft output is used (equivalent to `binarize_niftis = False` I believe). If the a roc_increment is provided than there will automatically be value and it implies we want to binarize. I feel like `binarize_niftis` is redundant with `binarize_niftis`. Let me know your thoughts, I might have forgotten a use case :)

Reply:
Explained [above](https://github.com/ivadomed/ivadomed/pull/383#issue-456074565):

> Output` niftis are binarized using binarize_predictions (which is the optimal threshold if the ROC analysis has ben run) if binarize_nifitis=true. This allows us to potentially compute metrics on hard predictions while keeping soft nifitis.

Which is exactly what we want to do in our current experiments: compute Dice on hard but keep nifitis soft for visualisation.

Reply:
This will create discrepancy between the output labels and metrics (metrics won't be representative of the outputs in pred_masks), personally I would keep a single parameter since visualization is more for debugging (and classic argument our config file is pretty populated). But either way I'm happy :)

Reply:
I hear your point :-) very valid.

But with allowing a soft output (while computing the metrics on hard masks, ie the right way) we can still binarise it further if we want (with a python script or fsleyes etc) whereas the other way does not allow this (it will be bin, I can't reconstruct the soft). These niftis, in ivadomed routine, are mainly for debugging / visualisation, not for "production" such as in SCT.

Not sure what's best here.

Reply:
can we pls discuss it in a meeting